application/service/factory/service_factory.go

```diff
@@ -139,7 +140,7 @@ func (f *ServiceFactory) OrganizationService() service.OrganizationService {
 }
 
 func (f *ServiceFactory) InvitationService() service.InvitationService {
-	return invitationservice.NewInvitationService(f.getContext(), f.config)
+	return invitationservice.NewInvitationService(f.getContext(), f.config, f.getContext().Services().WITService())
```

Comment:
If `WITService` is now available via the service layer, we don't need to pass in the parameter here - simply call `s.Services.WitService()` wherever it is needed.

Reply:
I should also add that this is the _proper_ way to do service to service interaction in the auth service, as going via the services interface gives us automatic transaction control (which is not needed in this case, however is a good practice to follow).

Reply:
@sbryzak we can simply call `s.Services.WitService()`, But in that case we can't assign `dummyWITService` for it in case of testing. this is why I am passing it as parameter.

Reply:
I understand, mocking the service is the concern. @dipak-pawar , could you explore the possibility of using a mock'd service without having to pass it as a parameter? We are gonna need that anyway soon in a many many places.

Reply:
@dipak-pawar , you could try doing a 
```
type TestGormDB struct {
    GormDB
}

func (testGormDB *TestGormDB) () service.WITService {
    return myMockWITService
}
```

Or, as @sbryzak suggested
do something like https://github.com/fabric8-services/fabric8-auth/blob/master/notification/service/notification.go#L42

Reply:
@sbryzak, We can use this as defined in [notification.go](https://github.com/fabric8-services/fabric8-auth/blob/master/notification/service/notification.go#L42). But with this way, you can't provide a dynamic mock of dependent services for that service. I would keep this simple for now. Maybe I will try to find more generic solution for this.

@sbose78 I am not sure about approach you suggested will work or not. I will try this in next PR when needed.